### ISSUE TITLE
Migrating from ImageChooserBlock to ImageBlock fails

### ISSUE THREAD

**Issue by Stormheg** [wagtail/wagtail#12514]
### Issue Summary

While I don't think we explicitly documented it, replacing instances of `ImageChooserBlock` with `ImageBlock` in your own blocks should be as straightforward as just swapping the blocks in your own block definitions. No data migration is necessary, as there is some [defensive coding in place to make this possible](https://github.com/wagtail/wagtail/blob/f1ee2f5c73056095cdfb9b1a6c72f936619bbe8c/wagtail/images/blocks.py#L151-L171)

While trialing Wagtail 6.3 release candidate 2 on the Wagtail.org codebase, I swapped a couple `ImageChooserBlock` with `ImageBlock` without a data migration as it should just work. See my branch: https://github.com/wagtail/wagtail.org/pull/478

I encountered a crash viewing the homepage though.

```
<snipped, for brevity>
  File "/Users/storm/projects/wagtail/wagtail.org/venv/lib/python3.13/site-packages/wagtail/blocks/stream_block.py", line 716, in _prefetch_blocks
    converted_values = child_block.bulk_to_python(raw_values.values())
  File "/Users/storm/projects/wagtail/wagtail.org/venv/lib/python3.13/site-packages/wagtail/blocks/struct_block.py", line 215, in bulk_to_python
    converted_values = child_block.bulk_to_python(raw_values)
  File "/Users/storm/projects/wagtail/wagtail.org/venv/lib/python3.13/site-packages/wagtail/images/blocks.py", line 173, in bulk_to_python
    struct_values = super().bulk_to_python(values)
  File "/Users/storm/projects/wagtail/wagtail.org/venv/lib/python3.13/site-packages/wagtail/blocks/struct_block.py", line 211, in bulk_to_python
    if name in val:
       ^^^^^^^^^^^
TypeError: argument of type 'NoneType' is not iterable
```

Here is some of the `ImageBlock` code referenced in the above stack trace

https://github.com/wagtail/wagtail/blob/f1ee2f5c73056095cdfb9b1a6c72f936619bbe8c/wagtail/images/blocks.py#L148-L175

Looking at the code paths referenced in above stack trace, we see we are skipping right past the defensive coding and it hitting `super().bulk_to_python(values)` which then tries and fails to do something with `values`.

Setting a breakpoint, I figured out `values = [None]` which really surprises me. This should be just `None` but somehow it is not? I am really not sure what the cause of this is, but I've seen it happen elsewhere in other client projects. I've always attributed it to weird migration issues and built defensive coding to work around it.

## Reproducing

Check out https://github.com/wagtail/wagtail.org/pull/478 and pull a copy of the staging database

## Suggested fix

I suggest changing the defensive coding to pick up on this weird condition as well. Unless there is a bigger picture issue I'm missing.

```diff
-if not values:
-    return []
+if not values or values[0] is None:
+    return []
```

### Technical details

-   Python version: Python 3.13.0
-   Django version: 5.0.9
-   Wagtail version: 6.3rc2
-   Browser version: Firefox 131

**Comment by Stormheg** [comment]
cc @Chiemezuo

**Comment by gasman** [comment]
Thanks @Stormheg! Replicated on bakerydemo as follows:

* Edit the `ImageBlock` in `bakerydemo/base/blocks.py` to have `image = ImageChooserBlock(required=False)`
* Edit a blog page, add an image block with caption filled in but image left blank, and publish
* Change `image` to a `wagtail.images.blocks.ImageBlock`:

```python
from wagtail.images.blocks import ImageBlock as WagtailImageBlock

class ImageBlock(StructBlock):
    image = WagtailImageBlock(required=False)
    caption = CharBlock(required=False)
    attribution = CharBlock(required=False)
```
* Revisit the edit view, receive `TypeError: argument of type 'NoneType' is not iterable`

**Comment by vossisboss** [comment]
Ha! Our bug hunt uncovered a bug. Thank you for chasing this down @Stormheg. And thank you @gasman for the quick fix suggestion.

**Comment by gasman** [comment]
Re the ability to directly replace `ImageChooserBlock` with `ImageBlock` not being documented - the sentence in [the release note](https://docs.wagtail.org/en/latest/releases/6.3.html#imageblock-with-alt-text-support) saying "This can be used as a direct replacement for `ImageChooserBlock`" was intended to communicate that. However, maybe that's being too subtle - it is a feature worth shouting about, after all :-)

Up to now I was inclined to avoid mentioning it in the main documentation for ImageBlock, as it's only relevant to people upgrading existing sites rather than learning about the block from the ground up - but again, maybe that's being too conservative. Will see what I can come up with, and I'm open to suggestions for how we can communicate this better.

**Comment by Chiemezuo** [comment]
Thanks for spotting this! @Stormheg 
And thanks for fixing this! @gasman

**Comment by vossisboss** [comment]
@gasman It might bear repeating at least once in the ImageBlock section of the documentation. I feel like the sentence you had in the release notes was pretty clear but some folks might eagerly click away to read about the new block before the point sinks in.

**Comment by Stormheg** [comment]
Thanks all!

**Comment by Stormheg** [comment]
FWIW; we worked on upgrading Wagtail.org to 6.3 again yesterday during office hours and determined the issue isn't resolved with Matthew's fix.

The original reproduction instructions still give the same crash, as the fix does not handle the case where `values = [None]`. Which - for reasons I do not entirely understand - is the case for Wagtail.org's database.

We worked on a fix together, the result being #12571.

I'm going to reopen this, as the issue is not fixed.
